src/meltano/core/logging/formatters.py

```diff
@@ -47,27 +57,15 @@ def rich_exception_formatter_factory(
     def _traceback(sio, exc_info) -> None:
         sio.write("\n")
         Console(file=sio, color_system=color_system, no_color=no_color).print(
-            Traceback.from_exception(*exc_info, show_locals=True)
+            Traceback.from_exception(
+                *exc_info,
+                show_locals=show_locals,
```

Comment:
Defaulting this to `False` means logs will be a bit less verbose and won't contain local variables of arbitrary functions. Users can still enable verbose tracebacks (see changes to the logging guide).

Reply:
@kgpayne I think you first made the call to default `show_locals` to `True`, so pinging you to make sure this looks good to you.

Reply:
@edgarrmondragon I think I was following examples from the Rich/structlog docs, so sure it wasn't a decision made with a strong use-case in mind. It made debugging more useful, which was the aim of the Rich formatting feature. However I appreciate, as a default, it shouldn't be used in prod. I know our docs suggest multiple logging configs, but this makes me think it would be good to associate logging features with environments somehow 🤔 I.e. By default in Meltano I want to see locals in dev, but not prod (without necessarily needing to configure my own logging configs). Just a thought 😅

Reply:
@kgpayne Thanks for the context!

> but this makes me think it would be good to associate logging features with environments somehow

Agreed. I recall folks requesting to be able to put the contents of `logging.yaml` in `meltano.yml`.

Reply:
Without having put much thought into it, being able to put the contents of `logging.yaml` into `meltano.yml` seems valuable.